libraries/AC_PID/AC_PID_2D.cpp

```diff
@@ -129,13 +129,16 @@ Vector2f AC_PID_2D::update_all(const Vector3f &target, const Vector3f &measureme
 //  If the limit is set the integral is only allowed to reduce in the direction of the limit
 void AC_PID_2D::update_i(const Vector2f &limit)
 {
-    Vector2f limit_direction = limit;
+    _pid_info_x.limit = false;
+    _pid_info_y.limit = false;
+
     Vector2f delta_integrator = (_error * _ki) * _dt;
-    if (!is_zero(limit_direction.length_squared())) {
+    if (!is_zero(limit.length_squared())) {
         // zero delta_vel if it will increase the velocity error
-        limit_direction.normalize();
-        if (is_positive(delta_integrator * limit)) {
+        if (is_positive(delta_integrator * limit.normalized())) {
```

Comment:
You can remove the .normalized() as it does not change the result.

Reply:
@lthall I was fairly sure that we didn't need the normalization, but I found it used in a `control.cpp` so I thought it was better to keep it. 

In that case I think we can remove these normalization too?
https://github.com/ArduPilot/ardupilot/blob/e7f1a273e518d386f325896a3233cca8847c9000/libraries/AP_Math/control.cpp#L61

https://github.com/ArduPilot/ardupilot/blob/e7f1a273e518d386f325896a3233cca8847c9000/libraries/AP_Math/control.cpp#L78

Reply:
Yes, that is correct.